src/Label.tsx

```diff
@@ -1,72 +1,100 @@
-  margin-bottom: -1px; // offsets the 1px border
-  color: ${get('colors.fg.muted')};
-  border: ${get('borderWidths.1')} solid ${get('colors.border.default')};
-  box-shadow: none;
-  ${borderColor};
-  background-color: transparent;
-`
+export type LabelProps = {
+  /** The color of the label */
+  variant?: LabelColorOptions
+  /** How large the label is rendered */
+  size?: LabelSizeKeys
+} & SxProp
 
-const sizeVariant = variant({
-  variants: {
-    small: {
-      fontSize: 0,
-      lineHeight: '16px',
-      padding: '0px 8px'
-    },
-    medium: {
-      fontSize: 0,
-      lineHeight: '20px',
-      padding: '0 8px'
-    },
-    large: {
-      fontSize: 0,
-      lineHeight: '24px',
-      padding: '0 12px'
-    },
-    // corresponds to StateLabel fontSize/lineHeight/padding
-    xl: {
-      fontSize: 1,
-      lineHeight: '16px',
-      padding: '8px 12px'
-    }
-  }
-})
+export type LabelColorOptions =
+  | 'default'
+  | 'primary'
+  | 'secondary'
+  | 'accent'
+  | 'success'
+  | 'attention'
+  | 'severe'
+  | 'danger'
+  | 'done'
+  | 'sponsors'
+
+type LabelSizeKeys = 'small' | 'large'
 
-const Label = styled.span<
-  {
-    variant?: 'small' | 'medium' | 'large' | 'xl'
-    dropshadow?: boolean
-    outline?: boolean
-  } & BorderColorProps &
-    SxProp
->`
-  display: inline-block;
-  font-weight: ${get('fontWeights.semibold')};
-  color: ${get('colors.fg.onEmphasis')};
-  border-radius: ${get('radii.3')};
-  background-color: ${get('colors.neutral.emphasis')};
+export const variants: Record<LabelColorOptions, BetterSystemStyleObject> = {
+  default: {
+    borderColor: 'border.default'
+  },
+  primary: {
+    borderColor: 'fg.default'
+  },
+  secondary: {
+    borderColor: 'border.muted',
+    color: 'fg.muted'
+  },
+  accent: {
+    borderColor: 'accent.emphasis',
+    color: 'accent.fg'
+  },
+  success: {
+    borderColor: 'success.emphasis',
+    color: 'success.fg'
+  },
+  attention: {
+    borderColor: 'attention.emphasis',
+    color: 'attention.fg'
+  },
+  severe: {
+    borderColor: 'severe.emphasis',
+    color: 'severe.fg'
+  },
+  danger: {
+    borderColor: 'danger.emphasis',
+    color: 'danger.fg'
+  },
+  done: {
+    borderColor: 'done.fg',
+    color: 'done.emphasis'
+  },
+  sponsors: {
+    borderColor: 'sponsors.fg',
+    color: 'sponsors.emphasis'
+  }
+}
 
-  &:hover {
-    text-decoration: none;
+const sizes: Record<LabelSizeKeys, BetterSystemStyleObject> = {
+  small: {
+    height: '20px',
+    padding: '0 7px' // hard-coded to align with Primer ViewCompnents and Primer CSS
+  },
+  large: {
+    height: '24px',
+    padding: '0 10px' // hard-coded to align with Primer ViewCompnents and Primer CSS
   }
+}
 
-  ${sizeVariant}
-  ${props => (props.dropshadow ? 'box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12)' : '')}
-  ${props => (props.outline ? outlineStyles : '')} // must be last to override other values
-  ${sx}
+const Label = styled.span<LabelProps>`
```

Comment:
Can we add a deprecation JS Doc annotation here please. [Example](https://github.com/primer/react/pull/1887/files#diff-328aed1393905cc95e834f271fabcfd7ebbd1e857a081af3024286543933a268R112) (you may need to give a few seconds to load properly).